packages/ra-ui-materialui/src/field/ReferenceArrayField.spec.tsx

```diff
@@ -301,4 +302,12 @@ describe('<ReferenceArrayField />', () => {
         await screen.findByText('tag:management');
         await screen.findByText('tag:design');
     });
+
+    it('should handle IDs of different types', async () => {
+        render(<DifferentIdTypes />);
+
+        expect(screen.findByText('artist_1')).not.toBeNull();
+        expect(screen.findByText('artist_2')).not.toBeNull();
+        expect(screen.findByText('artist_3')).not.toBeNull();
```

Comment:
```suggestion
        expect(await screen.findByText('artist_1')).not.toBeNull();
        expect(await screen.findByText('artist_2')).not.toBeNull();
        expect(await screen.findByText('artist_3')).not.toBeNull();
```

Don't forget to make sure your test actually fails when you provide wrong values too :wink: